src/onnx/parse_matmulnbits.cpp

```diff
@@ -67,7 +67,15 @@ struct parse_matmulnbits : op_parser<parse_matmulnbits>
                            ". Actual dims: " + to_string_range(args[1]->get_shape().lens()));
 
         std::vector<size_t> expected_scales_lens{n * n_blocks_per_col};
-        if(args[2]->get_shape().lens() != expected_scales_lens)
+
+        // Reshape anything larger than 1 dimension into a 1d tensor so we can check if we have the right amount of elements. 
+        auto scale_input = args[2];
+        if(scale_input->get_shape().lens().size() > 1)
+        {
+            scale_input = info.add_instruction(make_op("reshape", {{"dims", {scale_input->get_shape().elements()}}}), scale_input);
+        }
+
+        if(scale_input->get_shape().lens() != expected_scales_lens)
```

Comment:
Since this is just to check then we should just use `.elements` instead of inserting a reshape: `if(args[2]->get_shape().elements() != (n * n_blocks_per_col}))`

Reply:
Easy enough, but don't we need these to be in the correct 1d shape for the input?

Reply:
No, because the first thing we do is reshape it to a 2d tensor of`{n, -1}`(where -1 is the remaining elements).

Reply:
Also, you are not even using the reshaped instruction that is inserted, so there is no reason to add something to be always removed by DCE.